### PR TITLE
Add tests of utils.safe_apply_template function and refactor it

### PR DESCRIPTION
**Pull request by momotaro98**
Hi, yaufai.
untei project is very exciting! Let me contribute to this project.

In this PR, I focused on utils.safe_apply_template function. (It's so interesting for me.)
I added unit tests as `untei/test_utils.py` and also refactored `safe_apply_template` func.

### [Note] how to run test

To run tests in local, I believe you should remove `untei.` in https://github.com/yaufai/untei/blob/master/untei/utils.py#L4
I didn't modify it because it affects the product codes.
and run

```
$ python3 untei/test_utils.py
```

I think untei has an issue about reference. It seems untei in development refers to untei installed by `pip3 install untei`.

### CODE REVIEW

File: untei/constants.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def _path_concatinate(p1, p2):
 
 class Const:
     # arguments
-    FILES_PATH = sys.argv[1] + "/"
+    FILES_PATH = sys.argv[1] + "/" if len(sys.argv) > 1 else "./"
```
Comment (momotaro98) [inline untei/constants.py]: To run test as `python3 test_utils.py`, I have to avoid indexing error in `sys.argv[1]`. So I modified this line.

**Comment by yaufai** [comment]
Thank you for your contribution, momotaro-san.
I really appreciate your better sophisticated way of coding and prevention of potential bugs (especially risks of duplicated temporally keys). Just one thing, is it common to put test modules under the same directory as other sources (I am not very sure of mainstream of open source projects in python)?  I would more like putting test modules under `untei/untei/test` or `untei/test` (better preferred in my opinion).
Could you please reconsider the directory structure for a moment?

**Comment by momotaro98** [comment]
As you said, we should not put test codes in prod codes. So, I moved the test codes to `untei/tests`.
We can run the tests with
```
$ cd yaufai/untei
$ python -m unittest discover tests
```
In addition, above command doesn't matter in https://github.com/yaufai/untei/blob/master/untei/utils.py#L4 .
We don't have to remove `untei.` in `utils.py`.
I'm sorry my understanding was wrong.

Thank you.

**Comment by yaufai** [comment]
Thanks for the changes. Looks GREAT For Me!

**Pull request merged**